File: dot-net/src/LSP.MailBuilder/LSP.MailBuilder/Builder/IMailMessageBuilder.cs
using System.Net.Mail;

namespace LSP.MailBuilder.Builder
{
    public interface IMailMessageBuilder<in T>
    {
        IMailMessageBuilder<T> WithTo(string to);
        IMailMessageBuilder<T> WithSubject(string subject);
        IMailMessageBuilder<T> WithFrom(string from);
        MailMessage BuildMessage();
        IMailMessageBuilder<T> WithEntity(T entity);
    }
}

File: java/lsp-mail-builder/src/IMailMessageBuilder.java
public interface IMailMessageBuilder<T> {
    IMailMessageBuilder<T> WithTo(String to);
    IMailMessageBuilder<T> WithSubject(String subject);
    IMailMessageBuilder<T> WithFrom(String from);
    MailMessage BuildMessage();
    IMailMessageBuilder<T> WithEntity(T entity);
}